MapboxNavigation/NavigationView.swift

```diff
@@ -227,6 +228,35 @@ open class NavigationView: UIView {
         instructionsBannerView.delegate = delegate
         statusView.delegate = delegate
     }
+    
+    /**
+     Updates the instructions banner for a given `RouteProgress`.
+     */
+    public func update(for routeProgress: RouteProgress) {
+        let stepProgress = routeProgress.currentLegProgress.currentStepProgress
+        let distanceRemaining = stepProgress.distanceRemaining
+        
+        guard let visualInstructions = stepProgress.remainingVisualInstructions else { return }
+        
+        for visualInstruction in visualInstructions {
+            if stepProgress.distanceRemaining <= visualInstruction.distanceAlongStep || stepProgress.visualInstructionIndex == 0 {
+                
+                nextBannerView.hide()
+                lanesView.update(for: visualInstruction, currentStepProgress: routeProgress.currentLegProgress.currentStepProgress)
+                
+                if lanesView.isHidden {
+                    nextBannerView.update(for: routeProgress)
+                }
+                
+                instructionsBannerView.set(visualInstruction)
+                
+                stepProgress.visualInstructionIndex += 1
```

Comment:
We're updating this index now in two places?

https://github.com/mapbox/mapbox-navigation-ios/blob/a1d71e469df367e81733318b0c0375bf881ffda6/MapboxNavigation/InstructionsBannerView.swift#L114-L131

Reply:
@bsudekum nope, these are consumed by 2 different `viewcontrollers` -- `RouteMapViewController` and `CustomViewController`.

Reply:
This is incorrect. CustonViewController is purely an example of using InstructionsBannerView. RouteMapViewController uses it here:

https://github.com/mapbox/mapbox-navigation-ios/blob/245901d26d9b67fe2a05e2252e39744e402a1bee/MapboxNavigation/RouteMapViewController.swift#L20